ports/stm32/boards/WEACT_F411_BLACKPILL/board.json

```diff
@@ -0,0 +1,22 @@
+{
+    "deploy": [
+        "../PYBV10/deploy.md"
+    ],
+    "docs": "",
+    "features": [],
+    "images": [
+        "WEACTV20_F411.jpg"
+    ],
+    "mcu": "stm32f411",
+    "product": "WeAct F411 'blackpill'. Default variant is v3.1 with no SPI Flash.",
+    "thumbnail": "",
+    "url": "https://github.com/WeActStudio/WeActStudio.MiniSTM32F4x1",
+    "variants": {
+        "V13": "v1.3 board with no SPI Flash",
+        "V13_FLASH_4M": "v1.3 board with 4MB SPI Flash",
+        "V20_FLASH_4M": "v2.0 board with 4MB SPI Flash",
+        "V31_FLASH_8M": "v3.1 board with 8MB SPI Flash",
+        "V31_XTAL_8M": "v3.1 board with 8MHz crystal"
```

Comment:
Just to be clear, this 8MHz crystal version doesn't have SPI flash?

How do people distinguish v3.1 with the different crystals?

Reply:
I can only suggest the way nspsck does here:
https://github.com/nspsck/STM32F411CEU6_BlackPill_Micropython/blob/main/README.md

> For 8MHz crystal and 16MB spi flash:

make -j BOARD=WEACTF411CE BOARD_VARIANT=VARIANTS CRYSTAL_FREQ=8 SPI_FLASH_SIZE=16

For 25MHz crystal and 16MB spi flash, you have to leave CRYSTAL_FREQ out:

make -j BOARD=WEACTF411CE BOARD_VARIANT=VARIANTS SPI_FLASH_SIZE=16

maybe ignore BOARD_VARIANT depending on how Andrew builds these variants.

Reply:
So far there is only one variant with 8MHz xtal ... no external flash.

Reply:
To build a variant with this PR use:
```
make BOARD=WEACT_F411_BLACKPILL BOARD_VARIANT=V31_XTAL_8M
```

Reply:
Do I need to re-load any of the build files since `203a2a9`?

Reply:
You probably need to re-pull this PR.  But some of the comments here need addressing before it will build correctly for all variants.

Reply:
Thank you, I'll wait.

Reply:
> Just to be clear, this 8MHz crystal version doesn't have SPI flash?

According to the store pages: No. But the footprint for the flash chip is present, so people could add a flash chip. But then they have to adapt the build files and build the firmware themselves.

> How do people distinguish v3.1 with the different crystals?

They have to look at the crystal. 

I re-ordered a 8MHz crystal board and will get it next week. Then I can check whether the configuration works. ATM it does not look right, since `MICROPY_HW_CLK_PLLM` is defined unconditionally as 25 for all boards. But maybe that definition is not effective.